packages/vesting-base-lp/src/handlers.rs

```diff
@@ -0,0 +1,819 @@
+    if !amount.is_zero() {
+        msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: xyk_lp_token.to_string(),
+            msg: to_binary(&Cw20ExecuteMsg::Send {
+                contract: xyk_pair.to_string(),
+                amount,
+                msg: to_binary(&PairCw20HookMsg::WithdrawLiquidity { assets: vec![] })?,
+            })?,
+            funds: vec![],
+        }))
+    }
+    // push the next migration step as a callback message
+    msgs.push(
+        CallbackMsg::ProvideLiquidityToClPairAfterWithdrawal {
+            ntrn_denom,
+            ntrn_init_balance,
+            paired_asset_denom,
+            paired_asset_init_balance,
+            cl_pair,
+            slippage_tolerance,
+            user,
+        }
+        .to_cosmos_msg(&env)?,
+    );
+
+    Ok(Response::default().add_messages(msgs))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn provide_liquidity_to_cl_pair_after_withdrawal_callback(
+    deps: DepsMut,
+    env: Env,
+    ntrn_denom: String,
+    ntrn_init_balance: Uint128,
+    paired_asset_denom: String,
+    paired_asset_init_balance: Uint128,
+    cl_pair_address: Addr,
+    slippage_tolerance: Decimal,
+    user: VestingAccountResponse,
+) -> Result<Response, ContractError> {
+    let ntrn_balance_after_withdrawal = deps
+        .querier
+        .query_balance(env.contract.address.to_string(), ntrn_denom.clone())?
+        .amount;
+    let paired_asset_balance_after_withdrawal = deps
+        .querier
+        .query_balance(env.contract.address.to_string(), paired_asset_denom.clone())?
+        .amount;
+
+    // calc amount of assets that's been withdrawn
+    let withdrawn_ntrn_amount = ntrn_balance_after_withdrawal.checked_sub(ntrn_init_balance)?;
+    let withdrawn_paired_asset_amount =
+        paired_asset_balance_after_withdrawal.checked_sub(paired_asset_init_balance)?;
+
+    let mut msgs: Vec<CosmosMsg> = vec![];
+
+    if !withdrawn_ntrn_amount.is_zero() && !withdrawn_paired_asset_amount.is_zero() {
+        msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: cl_pair_address.to_string(),
+            msg: to_binary(&PairExecuteMsg::ProvideLiquidity {
+                assets: vec![
+                    native_asset(ntrn_denom.clone(), withdrawn_ntrn_amount),
+                    native_asset(paired_asset_denom.clone(), withdrawn_paired_asset_amount),
+                ],
+                slippage_tolerance: Some(slippage_tolerance),
+                auto_stake: None,
+                receiver: None,
+            })?,
+            funds: vec![
+                Coin::new(withdrawn_ntrn_amount.into(), ntrn_denom),
+                Coin::new(withdrawn_paired_asset_amount.into(), paired_asset_denom),
+            ],
+        }))
+    }
+
+    msgs.push(CallbackMsg::PostMigrationVestingReschedule { user }.to_cosmos_msg(&env)?);
+
+    Ok(Response::default().add_messages(msgs))
+}
+
+fn post_migration_vesting_reschedule_callback(
```

Comment:
as far as i can see. We reschedule remainded tokens over the same vesting period `end of lockdrop` - `end of lockdrop + 3m`. It gaves extra "cliff" to the users. Lets say: 
1) originally we created a schedule 10 coins from day 1 to day 10.
2) at day 5 user claims 5 vested coins
3) we run the migrations routine (lp convertation is 1 to 1)
4) we vest 5 new coins over days 1 to 10. 
5) Today is day 5 and user can claim 2.5 extra coins right after migration.

Reply:
Migration to new the contractes happens after september 5th(vesting ends), and there are no unvested tokens when migrations happens. So we do not care about the schedules transformation.